distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1692,4 +1913,85 @@ private CometDClient setupCometDClient(List<String> channels) throws Exception {
     protected Option[] configureCustom() {
         return options(mavenBundle("ddf.test.thirdparty", "restito").versionAsInProject());
     }
+
+    /**
+     * Sets up a download response and starts a download for testing in progress downloads.
+     */
+    private class DownloadResponse {
```

Comment:
If this is an inner class, it might be easier to make it a `static` inner class so we don't depend on the state of the parent and have to do ugly construction (i.e. `new Parent( ).new Child( )`)